packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_token_burn_token.test.ts

```diff
@@ -0,0 +1,190 @@
+    const script = await providers.elliptic.script()
+    const promise = builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(-2) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: {
+          stack: []
+        }
+      }
+    }, script)
+
+    await expect(promise).rejects.toThrow('The value of "value" is out of range. It must be >= 0 and <= 4294967295. Received -200000000')
+  })
+
+  it('should throw an error if not enough tokens are available to burn', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(15) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: {
+          stack: []
+        }
+      }
+    }, script)
+
+    // Ensure the created txn is correct
+    const promise = sendTransaction(testing.container, txn)
+
+    await expect(promise).rejects.toThrow(DeFiDRpcError)
+    await expect(promise).rejects.toThrow("DeFiDRpcError: 'BurnTokenTx: amount 0.00000000 is less than 15.00000000 (code 16)', code: -26")
+  })
+
+  it('should burnToken without context', async () => {
+    // Fund 100 BTC TOKEN
+    await testing.token.send({
+      address: await providers.getAddress(),
+      amount: 100,
+      symbol: 'BTC'
+    })
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+
+    const txn = await builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(15) }],
+      burnType: 0,
+      from: script,
+      variantContext: {
+        variant: 0,
+        context: script
+      }
+    }, script)
+
+    // Ensure the created txn is correct
+    await sendTransaction(testing.container, txn)
+    await testing.generate(1)
+
+    const attributes = await testing.rpc.masternode.getGov(attributeKey)
+    const burntKeyRegex = /^v0\/live\/economy\/consortium_members\/\d+\/\d+\/burnt$/
+    const keys: string[] = Object.keys(attributes.ATTRIBUTES)
+
+    // Verify the token is not burned to any consortium
+    expect(keys.every(key => burntKeyRegex.exec(key) === null)).toStrictEqual(true)
+
+    // Verify the token is deducted
+    const accAfter = await testing.rpc.account.getAccount(await providers.getAddress())
+    expect(accAfter).toStrictEqual(['85.00000000@BTC'])
+  })
+
+  it('should burnToken with context', async () => {
+    // Fund 100 BTC TOKEN
+    await testing.token.send({
+      address: await providers.getAddress(),
+      amount: 100,
+      symbol: 'BTC'
+    })
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const wavesColScript = P2WPKH.fromAddress(RegTest, wavesConsortiumAddress, P2WPKH).getScript()
+
+    const txn = await builder.tokens.burnTokens({
+      amounts: [{ token: Number(idBTC), amount: new BigNumber(15) }],
```

Comment:
```suggestion
      amounts: [{ token: Number(idBTC), amount: new BigNumber(30) }],
```

Use a different number for this test for variability.